script.xbmc.boblight/default.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*- 
+'''
+    Boblight for Kodi
+    Copyright (C) 2012 Team XBMC
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+import xbmc
+import xbmcaddon
+import xbmcgui
+import os
+
+__addon__      = xbmcaddon.Addon()
+__cwd__        = __addon__.getAddonInfo('path')
+__scriptname__ = __addon__.getAddonInfo('name')
+__version__    = __addon__.getAddonInfo('version')
+__icon__       = __addon__.getAddonInfo('icon')
+__ID__         = __addon__.getAddonInfo('id')
+__language__   = __addon__.getLocalizedString
+
+__profile__    = xbmc.translatePath( __addon__.getAddonInfo('profile') )
+__resource__   = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) )
+
+sys.path.append (__resource__)
+
+from settings import *
+from tools import *
+
+log( "[%s] - Version: %s Started" % (__scriptname__,__version__))
+
+capture_width  = 32
+capture_height = 32
+settings       = settings()
+capture        = xbmc.RenderCapture()
+useLegacyApi   = True
+
+class MyPlayer( xbmc.Player ):
+  def __init__(self, monitor):
+    xbmc.Player.__init__( self )
+    self.monitor = monitor
+    self.playing = False
+    log('MyPlayer - init')
+        
+  def onPlayBackStopped( self ):
+    self.playing = False
+    myPlayerChanged('stop', self.monitor)
+
+  def onPlayBackPaused( self ):
+    myPlayerChanged('paused', self.monitor)
+
+  
+  def onPlayBackEnded( self ):
+    self.playing = False
+    myPlayerChanged('stop', self.monitor)
+  
+  def onPlayBackStarted( self ):
```

Comment:
I think you need onAvStarted here, you're starting the rendercapture before the stream is actually available

Reply:
Ok, I can do that. `onAVStarted` seems to be a relatively recent API (v18), so I think that's why it wasn't used. `RenderCapture` doesn't seem to mind being started early.